schema.ts

```diff
@@ -1232,6 +1232,82 @@ export interface ObjectSchemaDefinition<K extends string, L extends string>
   // TODO(dweitzman): Only support options in the typing when the codaType is ValueHintType.SelectList.
 }
 
+/**
+ * Discriminent used to determine the permission type
+ * @hidden
+ */
+export enum PrincipalType {
+  User = 'user',
+  Group = 'group',
+  Domain = 'domain',
+  Public = 'public',
+}
+
+/**
+ * This type definition represents properties required for all permissions.
+ *
+ * TODO(sam): Unhide this
+ * @hidden
+ */
+interface BasePermission {
```

Comment:
Was looking at how we model this in the public API, this object is more accurately a BasePrincipal, and the others are UserPrincipal, GroupPrincipal, etc. A _permission_ is a principal joined with an access level (can view, can edit, etc). We may not bother with access levels at this point while things are readonly, but we should use that more accurately vocabulary.

Reply:
Ah, I skimmed past the row id. I think we should model this in a way where permissions are self-contained objects. The executeGetPermissions return value can pair a permission with an object id. But I don't think that implies they should be part of the same object.

```ts
interface ReturnValueNeedingAGoodName {
  permissions: Permission[];
  rowId: string | number;
}

type Principal = UserPrincipal | GroupPrincipal | AnyonePrincipal;

interface Permission {
  principal: Principal;
  // Later perhaps
  // access: Access;  // CanView, CanEdit, CanComment, whatever
}
```

Reply:
Updated this!